R/generate_cpp.R

```diff
@@ -412,6 +412,46 @@ generate_state_types <- function(
   invisible(dest)
 }
 
+#' Generate C++ for state saver types
+#'
+#' @param dest The destination to write generated code to.
+#' @param output_csv Path to csv file with output specification in it
+#'
+#' @return Nothing, called to generate code in src dir
+#' @keywords internal
+generate_state_saver_types <- function(
```

Comment:
ive seen the start of this function in a couple of places, mainly cos it's just doing basic parsing, i think this can go into one function, with the header generation as well as these happen for all these generation functions, we should be able to just attach the `header` variable to the parent env inside a function easily, i think the call to the cpp can also be extracted so we are just providing one function that assigns the specific thing that it needs

perhaps will tell you on call!

Reply:
Thanks, that sounds good. I've moved header into the `generate_cpp` function, and updated that to write out the file too. 

I've also put the input and output reading into a util which we can then call from all the generate function. See `read_output_csv` and `read_input_csv`. Let me know what you think